File: solutions/cpp/char_frequency.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

unordered_map<char,int> charFrequency(string s) {
    unordered_map<char,int> f;
    for(char c:s) f[c]++;
    return f;
}

int main() {
    auto m = charFrequency("banana");
    for(auto &x:m) cout<<x.first<<":"<<x.second<<" ";
}

File: solutions/java/char-frequency.java
import java.util.*;

public class CharFrequency {
    public static Map<Character,Integer> charFrequency(String s) {
        Map<Character,Integer> freq = new HashMap<>();
        for (char ch : s.toCharArray()) freq.put(ch, freq.getOrDefault(ch,0)+1);
        return freq;
    }
    public static void main(String[] args) {
        System.out.println(charFrequency("banana"));
    }
}